### PR TITLE
fix: Improve the legend for the different plots

### LINKED ISSUE #1487: enh: Improve the information shown in the legend of the metrics displays
### What would you like to say?

Looking more at the legends of the metric display, I think that we could simplify it by choosing the information that we want to display. For instance, we have redundant information in the label (e.g. `data_source`) and we could find a solution by showing at a single location (e.g. title).

### PR DESCRIPTION
**Pull request by glemaitre**
closes #1487 

This PR intends to better place by default the location of the legend based on some good heuristic.

In short, we reduce the amount of redundant information in the legend to minimize its size and place it in the best possible location.

I put an example below that allow to play with the different configurations.

<details>

```python
# %% [markdown]
# # Binary classification

# %%
from sklearn.datasets import make_classification

X, y = make_classification(n_samples=1_000, n_classes=2, random_state=0)

# %% [markdown]
# ## `EstimatorReport`

# %%
from skore import EstimatorReport
from sklearn.linear_model import LogisticRegression

report = EstimatorReport(LogisticRegression(), X_train=X, y_train=y, X_test=X, y_test=y)

# %%
report.metrics.roc(data_source="train").plot()

# %% [markdown]
# ## `CrossValidationReport`

# %%
from skore import CrossValidationReport

# cv_splitter will change the location of the legend
report = CrossValidationReport(
    LogisticRegression(max_iter=10_000), X=X, y=y, cv_splitter=10
)

# %%
report.metrics.precision_recall().plot()

# %% [markdown]
# ## `ComparisonReport`

# %%
from skore import ComparisonReport

# The number of reports will change the location of the legend
report_1 = EstimatorReport(
    LogisticRegression(), X_train=X, y_train=y, X_test=X, y_test=y
)
report_2 = EstimatorReport(
    LogisticRegression(C=1e-10), X_train=X, y_train=y, X_test=X, y_test=y
)
comp_report = ComparisonReport({"report_1": report_1, "report_2": report_2})

# %% [markdown]
# # Multiclass classification

from sklearn.datasets import make_classification

# n_classes will have an impact on the location of the legend
X, y = make_classification(
    n_samples=1_000,
    n_classes=7,
    n_clusters_per_class=1,
    random_state=0,
    n_informative=10,
)

# %% [markdown]
# ## `EstimatorReport`

# %%
report = EstimatorReport(LogisticRegression(), X_train=X, y_train=y, X_test=X, y_test=y)
report.metrics.precision_recall().plot()

# %% [markdown]
# ## `CrossValidationReport`

# %%
report = CrossValidationReport(
    LogisticRegression(max_iter=10_000), X=X, y=y, cv_splitter=10
)
report.metrics.precision_recall().plot()


# %% [markdown]
# ## `ComparisonReport`

# %%
report_1 = EstimatorReport(
    LogisticRegression(), X_train=X, y_train=y, X_test=X, y_test=y
)
report_2 = EstimatorReport(
    LogisticRegression(C=1e-10), X_train=X, y_train=y, X_test=X, y_test=y
)
comp_report = ComparisonReport({"report_1": report_1, "report_2": report_2})

# %% [markdown]
# # Regression

# %%
from sklearn.datasets import make_regression

X, y = make_regression(random_state=10)

# %% [markdown]

# %%
from sklearn.linear_model import LinearRegression

report = EstimatorReport(LinearRegression(), X_train=X, y_train=y, X_test=X, y_test=y)
# kind="actual_vs_predicted" in `plot` will change the location of the legend
report.metrics.prediction_error().plot()

# %% [markdown]
# ## `CrossValidationReport`

# %%
report = CrossValidationReport(LinearRegression(), X=X, y=y, cv_splitter=10)
report.metrics.prediction_error().plot()

# %% [markdown]
# ## `ComparisonReport`

# %%
# the number of reports will have an impact on the location of the legend
reports = {
    f"report_{i}": EstimatorReport(
        LinearRegression(), X_train=X, y_train=y, X_test=X, y_test=y
    )
    for i in range(1, 7)
}
comp_report = ComparisonReport(reports)

# %%
comp_report.metrics.prediction_error().plot()

```

</details>

### CODE REVIEW

File: skore/tests/unit/sklearn/plot/test_roc_curve.py [hunk 1]
```diff
@@ -671,3 +670,136 @@ def test_roc_curve_display_wrong_report_type(pyplot, binary_classification_data)
...
+        bbox = legend.get_window_extent().transformed(display.ax_.transAxes.inverted())
+        assert 0 <= bbox.x0 <= 1
+    else:
+        bbox = legend.get_window_extent().transformed(display.ax_.transAxes.inverted())
+        assert bbox.x0 >= 1
+
+
+def test_roc_curve_display_legend_estimator_report(
+    pyplot, binary_classification_data, multiclass_classification_data
+):
```
Comment (augustebaum) [inline skore/tests/unit/sklearn/plot/test_roc_curve.py:691]: can you split this test into several?

**Review by auguste-probabl — CHANGES_REQUESTED** [review]
Looks good apart from splitting tests

**Comment by thomass-dev** [comment]
[automated comment] Please update your PR with main, so that the `pytest` workflow status will be reported.

**Review by glemaitre — COMMENTED** [review]
Thanks @auguste-probabl to have finished this PR. Let's merge and iterate once with have `subplots`
